### PR TITLE
Fix rethrow on strings

### PR DESCRIPTION
**Pull request by shvaikalesh**
Replaces https://github.com/brunch/brunch/pull/1577

### CODE REVIEW

File: lib/utils/error.js [hunk 1]
```diff
@@ -2,9 +2,14 @@
...
-    .replace(/#{\s*(\w+)\s*}/g, (_, key) => params[key] || '')
-    .concat(`\n${params.error || ''}`);
+  let withError = false;
+  let message = template.replace(/#{\s*(\w+)\s*}/g, (_, key) => {
+    if (key === 'error') withError = true;
+    return params[key] || '';
+  });
+
+  if (!withError) message += `\n${params.error || ''}`;
+  return message;
```
Comment (denysdovhan) [inline lib/utils/error.js]: Why not …?

```js
const format = (template, params) => {
  let withError = false;
  return template
    .replace(/#{\s*(\w+)\s*}/g, (_, key) => {
      if (key === 'error') withError = true;
      return params[key] || '';
    })
    .concat(`\n${!withError ? (params.error || '') : ''}`)
};
```

Comment (denysdovhan) [inline lib/utils/error.js]: Hmm, actually ```.concat(`\n${!withError ? (params.error || '') : ''}`)``` looks not so obvious.

File: lib/utils/error.js [hunk 2]
```diff
@@ -14,16 +13,11 @@ const format = (template, params) => {
...
 class BrunchError extends Error {
   constructor(code, params) {
-    const message = code in messages ?
-      format(messages[code], params || {}) :
-      pipelineMessages.includes(code) ? params.error.message : code;
-
+    const message = format(messages[code] || code, params || {});
     super(message);
 
-    Error.captureStackTrace(this, this.constructor);
```
Comment (denysdovhan) [inline lib/utils/error.js]: Why don't we capture stack trace here?

Comment (shvaikalesh) [inline lib/utils/error.js]: because we `extends`

File: lib/utils/error.js (hunk 1 above)
Comment (shvaikalesh) [inline lib/utils/error.js]:
```js
`\n${!withError ? (params.error || '') : ''}`
```

meh

File: lib/utils/error.js (hunk 2 above)
Comment (denysdovhan) [inline lib/utils/error.js]: right, this should work automatically

File: lib/utils/error.js (hunk 1 above)
Comment (shvaikalesh) [inline lib/utils/error.js]: When people see `.concat` they usually think of arrays, not strings. So I would prefer `+=`.

**Pull request merged**